### ISSUE TITLE
[BUG]: Anime full season search doesn't work

### ISSUE THREAD

**Issue by movianlost** [elgatito/script.elementum.burst#265]
When I try to search an anime season, Burst doesn't return any results. This is because in providers.json there is anime_keywords and season_keywords.

This results in a conflict because this is a season search while also having 'anime':'true', and there is no definition on how to search a season for anime there is only for tvshows.

The solution is to add anime_season_keywords that will define the query for searching an anime season.

**Comment by elgatito** [comment]
@movianlost the only difference between anime episode and usual episode searches is the usage of absolute number in the search query.
For anime seasons it should be the usual season search.

**Comment by movianlost** [comment]
I was doing some testing and it seems like I can only search for episodes that exists in the first season. If I click on any episode on season 1 it works fine,however,  If I go to season 2 and click on one of the episodes it starts and stops immediately without any search, if I go back and search the whole season 1, then it start burst and keeps searching until timeout and doesn't return any data.

This what I did for testing: 

- I clicked on Naruto S1E1 -> Worked fine.
- Searched Naruto S1 -> it stared searching and stopped after timeout wihout any result, this happen to all seasons, this only happens to anime, other tv shows work fine for season search.
- I clicked on Naruto S2E1 -> Started and Stopped immediately without any results, I noticed from the log that that it uses /season/2/episode/53 whereas I click on episode 1, I know anime uses absolute numbers, but I don't think this is the intended behavior as it tries to search for episode 53 inside season 2.

Here is the log  from the moment I started the above testing:
https://pastebin.com/raw/k5FtSyY3

**Comment by movianlost** [comment]
I think I found why S2E1 doesn't play, this is because if you call tmdb api for season 2, it will return the list of episodes available for that season, the first episode there has key "episode_number" with a value 53, so the returned data is correct.

The problem is that when you request the below link with Elementum, it returns an error.
`/show/46260/season/2/episode/53/links/ナルト S02E53?doresume=false`

I'm not sure but I suspect Elementum is calling another API and requesting some info for season 2 episode 53 and that API is using a different naming scheme for anime episodes which results in not being able to find the item.

Can you check if Elementum is calling TVDB API with the above data? this maybe the cause of the issue.

**Comment by elgatito** [comment]
@movianlost TMDB does NOT contain absolute episode number information. To get one, we use TVDB api. And still we depend on having TVDB ID inside TMDB information, and absolute episode number for specific episode in TVDB information.

> /show/46260/season/2/episode/53/links/ナルト S02E53?doresume=false

*ALL* calls to Elementum use TMDB numbering, they are never absolute. Absolute number is only sent to search providers and to search a file into multi-file torrents.

**Comment by elgatito** [comment]
> Here is the log from the moment I started the above testing:
> https://pastebin.com/raw/k5FtSyY3

Your log is private. It is not opening.

**Comment by movianlost** [comment]
It's Pastebin issue, it doesn't open for me too even while logged-in . I upload the log to Github, here is the raw file logs:

https://gist.githubusercontent.com/movianlost/cb010365f893e674be4e356cc4d6c852/raw/806970e5844550be35b48c29fef114c2b3ed0ecb/gistfile1.txt

> ALL calls to Elementum use TMDB numbering, they are never absolute. Absolute number is only sent to search providers and to search a file into multi-file torrents.

In the example that I tried where I click on Naruto S02E01, TMDB have this episode as E02E53 which is fine, then Elementum take this data and tries to use it with TVDB and it gets an error because S02E53 doesn't exist in TVDB while it exists in TMDB.

You can't have two information providers like TMDB and TVDB and expect them to agree on the episode number, one will use DVD release episode numbers, the other may use TV release , you can never be sure.

Since TVDB have already gone to a subscription model where either the developer have to pay thousands of dollars yearly or the user have to buy $12/year to get their api key, I suggest we remove TVDB and focus solely on TVDB API, this way there won't be conflict of information between TMDB and TVDB.


> TMDB does NOT contain absolute episode number information. To get one, we use TVDB api. And still we depend on having TVDB ID inside TMDB information, and absolute episode number for specific episode in TVDB information.

When you call TMDB api for a certain show, it will return the number of episodes in each season, using this information and simple calculation you get the absolute episode number.

For example if we have a show with season:episode_numbers like this {s1:30,s2:40,s3:40,s4:30} and you want to convert S03E10 to absolute number you do 30+40+10=80

With this we avoid any conflict between TMDB and TVDB.

**Comment by movianlost** [comment]
@elgatito

In Elementum Settings, under Providers>>Search there are options like 'use original title', 'use english title as original for anime search' and 'use earliest date as basic for searching'.

Can you add a 4th option and call it something like 'Treat anime as normal series'? if the user enables this, Elementum will not check if the serie is an anime and will treat it like it treats any other tv show.

**Comment by elgatito** [comment]
@movianlost v0.0.64 now correctly detects season search and treat as usual season search, not anime episode search.

**Comment by movianlost** [comment]
@elgatito  Great, thank you, I knew it was checking if the show was Anime even when performing a season search.

Can you add an option in Elementum next update under   `Addon-on settings>>Providers>>Search` named ' named something like `Treat anime like normal series` that doesn't change episode number to absolute number when sending it to the Burst providers ?
In my situation, I'm adding a provider in Burst that must have the same season and episode number as in TMDB.

**Comment by elgatito** [comment]
@movianlost It will not work as you expect it to. 
Try the latest Elementum version, there is a change for tmdb episode numbering.

**Comment by movianlost** [comment]
@elgatito I'll test the latest version with multiple shows and report back.

Can you tell me why it won't work as I expect? 

Can you tell me what are the kind of data Elementum gets from TVDB besides absolute numbers for anime?

**Comment by movianlost** [comment]
@elgatito I've tested the latest version and it fixed the shows the didn't work before. But I still have some issues. I've opened an issue in the `plugin.video.elementum` repository because I feel its an Elementum's not Burst's issue.

Here is the issue https://github.com/elgatito/plugin.video.elementum/issues/736

This same seasons opens fine using TheMovieDB Helper and data is for that season is available through TMDB api, but it still doesn't work.
